File: golang/uPIMulator/src/linker/kernel/instruction/cc/no_cc.go
package cc

import (
	"errors"
)

type NoCc struct {
	condition Condition
}

func (this *NoCc) Init(condition Condition) {
	conditions := map[Condition]bool{}

	if _, found := conditions[condition]; !found {
		err := errors.New("condition is not allowed")
		panic(err)
	}

	this.condition = condition
}

func (this *NoCc) Condition() Condition {
	return this.condition
}

File: python_cpp/uPIMulator_frontend/src/abi/isa/cc/no_cc.py
from typing import Set

from abi.isa.cc._base_cc import BaseCC
from abi.isa.instruction.condition import Condition


class NoCC(BaseCC):
    def __init__(self, condition: Condition):
        super().__init__(NoCC.conditions(), condition)

    @staticmethod
    def conditions() -> Set[Condition]:
        return set()
